client/src/components/ComboBox/ComboBox.tsx

```diff
@@ -153,12 +153,11 @@ export default function ComboBox<ComboBoxOption extends ComboBoxItem>({
       setInputItems(filtered);
       // Always reset the first item to highlighted on filtering, to speed up selection.
       setHighlightedIndex(0);
+      // Hide any preview when the user starts typing.
+      setPreviewedIndex(undefined);
```

Comment:
I spent a lot of time on this wondering what was the better behavior. In the end I thought being more aggressive about hiding the preview was probably the safer option.